xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/DefaultEntityResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.xml.internal;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.xml.EntityResolver;
+
+/**
+ * Uses css4j's entity resolver.
```

Comment:
I think it would be good to explain why we use CSS4's resolver. I'd mention avoiding code duplication, and that it protects XWiki from most information leaks, DoS attacks based on entity expansion/recursion like the 'billion laughs attack' (copied from https://css4j.github.io/api/2.1/io/sf/carte/doc/xml/dtd/DefaultEntityResolver.html).

Reply:
Pushed a new version of the commit explaining why. I wasn't much verbose because you wanted some confidentiality (and Github PRs are quite visible).